homework/index.js

```diff
@@ -30,13 +30,175 @@
     return elem;
   }
 
+  function createHtml(jsonData) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('div', root, { id: 'header' });
+    createAndAppend('img', header, { src: './hyf.png', alt: 'HackYourFuture logo' });
+    createAndAppend('h1', header, { text: 'HackYourFuture Github Repositories' });
+    const choice = createAndAppend('div', root, { id: 'choice' });
+    createAndAppend('h3', choice, { text: 'Please Select a Repository Below' });
+    const selection = createAndAppend('div', choice, { id: 'selection' });
+    createAndAppend('p', selection, { text: 'HYF Repositories: ' });
+    const selectElem = createAndAppend('select', selection, { id: 'select' });
+    const repos = jsonData;
+    const repoNames = repos
+      .map(repo => repo.name)
+      .sort((x, y) => (String(x).toUpperCase() > String(y).toUpperCase() ? 1 : -1));
+
+    repoNames.forEach((repoName, index) => {
+      createAndAppend('option', selectElem, { text: repoName, value: index });
+    });
+
+    const contentDiv = createAndAppend('div', root, { id: 'content' });
+    const leftDiv = createAndAppend('div', contentDiv, { id: 'left-side' });
+    const rightDiv = createAndAppend('div', contentDiv, { id: 'right-side' });
+
+    function getSelectedOption(sel) {
+      let opt;
+      for (let i = 0; i < sel.options.length; i++) {
+        opt = sel.options[i];
+        if (opt.selected === true) {
+          break;
+        }
+      }
+      return opt;
+    }
+
+    function getSelectedIndex(name, opt) {
+      let selectedIndex = 0;
+      for (let i = 0; i < name.length; i++) {
+        if (name[i].name === opt.text) {
+          selectedIndex = i;
```

Comment:
If you write `return i;` here, and `return 0;` in line 74, you break the function early. Because when the index has been found, you can stop the loop!